lnbits/core/views/api.py

```diff
@@ -246,8 +244,13 @@ async def api_payments_create(
         return await api_payments_pay_invoice(
             invoiceData.bolt11, wallet.wallet
         )  # admin key
-    # invoice key
-    return await api_payments_create_invoice(invoiceData, wallet.wallet)
+    elif not invoiceData.out:
+        # invoice key
+        return await api_payments_create_invoice(invoiceData, wallet.wallet)
+    else:
+        raise HTTPException(
+            status_code=HTTPStatus.BAD_REQUEST, detail="Key type is invalid"
+        )
```

Comment:
This is the error for the edge case mentioned in the OP.

Reply:
Nitpick: Maybe it'd be more informative to the user with error message detail="Requires invoice key" instead of detail="Key type is invalid"